app/src/main/java/org/wikipedia/createaccount/CreateAccountActivity.kt

```diff
@@ -289,19 +295,21 @@ class CreateAccountActivity : BaseActivity() {
         }
 
         override fun run() {
-            disposables.add(ServiceFactory.get(wiki).getUserList(userName)
-                    .subscribeOn(Schedulers.io())
-                    .observeOn(AndroidSchedulers.mainThread())
-                    .subscribe({ response ->
-                        response.query?.getUserResponse(userName)?.let {
-                            binding.createAccountUsername.isErrorEnabled = false
-                            if (it.hasBlockError) {
-                                handleAccountCreationError(it.error)
-                            } else if (!it.canCreate) {
-                                binding.createAccountUsername.error = getString(R.string.create_account_name_unavailable, userName)
-                            }
-                        }
-                    }) { L.e(it) })
+            lifecycleScope.launch(CoroutineExceptionHandler { _, throwable ->
+                L.e(throwable)
+            }) {
+                val response = withContext(Dispatchers.IO) {
+                    ServiceFactory.get(wiki).getUserList(userName)
+                }
+                response.query?.getUserResponse(userName)?.let {
+                    binding.createAccountUsername.isErrorEnabled = false
+                    if (it.hasBlockError) {
+                        handleAccountCreationError(it.error)
+                    } else if (!it.canCreate) {
+                        binding.createAccountUsername.error = getString(R.string.create_account_name_unavailable, userName)
+                    }
+                }
+            }
```

Comment:
Would it be possible to move this to the `ViewModel` class?

Reply:
Sure! I’ve moved the logic to the ViewModel and also removed the Runnable in the latest commit.